pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -983,6 +1001,31 @@ type WorkflowStatus struct {
 
 	// StoredWorkflowSpec stores the WorkflowTemplate spec for future execution.
 	StoredWorkflowSpec *WorkflowSpec `json:"storedWorkflowTemplateSpec,omitempty" protobuf:"bytes,14,opt,name=storedWorkflowTemplateSpec"`
+
+	// Synchronization stores the status of synchronization locks
+	Synchronization *SynchronizationStatus `json:"synchronization,omitempty" protobuf:"bytes,15,opt,name=synchronization"`
+}
+
+type SemaphoreStatus struct {
+	// Holding stores the list of resource acquired synchronization lock for workflows
```

Comment:
I think the SemaphoreStatus might need to be reworked. 

Currently, the holding list is a map of the semaphore key, to the holders list (i.e):

```yaml
status:
  synchronization:
    semaphores:
      holding:
        argo/configmap/semaphore-one/count:
          holders:
          - workflow-aaa-xx6rp-109132273
          - workflow-aaa-xx6rp-1777408785
```

But I think the following will follow K8s API conventions better, since it avoids a map with a user-defined, magic keys.

```yaml
status:
  synchronization:
    semaphores:
      holding:
      - semaphore: argo/configmap/semaphore-one/count
        holders:
        - workflow-aaa-xx6rp-109132273
        - workflow-aaa-xx6rp-1777408785
```